models/sipnet/R/write.configs.SIPNET.R

```diff
@@ -102,8 +102,12 @@ write.config.SIPNET <- function(defaults, trait.values, settings, run.id, inputs
   jobsh <- gsub("@BINARY@", settings$model$binary, jobsh)
   jobsh <- gsub("@REVISION@", settings$model$revision, jobsh)
   
-  jobsh <- gsub("@CPRUNCMD@", cpruncmd, jobsh)
-  jobsh <- gsub("@CPOUTCMD@", cpoutcmd, jobsh)
+  if(!is.identical(cpruncmd, character(0)) {
+    jobsh <- gsub("@CPRUNCMD@", cpruncmd, jobsh)
```

Comment:
Couple questions:
* are we sure that if these variables aren't defined in the pecan.xmxl that they'll be set to character(0)?
* If these variables don't exist, are we sure the job.sh will run correctly with the text of these tags left in the shell script? looking at the template job, it seems like they should be removed (i.e. gsubbed with an empty string)
https://github.com/PecanProject/pecan/blob/456433715edc46e9f06319cb66771f2f90a03697/models/template/inst/template_geo.job#L46

If the right thing to do is gsub with an empty string, and the is.identical is assuming the default is an empty string, this begs the question of why the default case is failing

Reply:
@meetagrawal09 I agree with your diagnosis, but I think the issue should be fixed on lines 76-77 rather than here. Maybe that could be something like `cpruncmd <- gsub("@OUTDIR@", settings$host$rundir, settings$host$rabbitmq$cpfcmd %||% "")`, then the replacements on lines 105-106 remain unconditional?